From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 2 Aug 2026 12:15:20 +0200
Subject: [PATCH 079/117] arm64: defconfig: enable D111 battery charging

Enable the SN2400 MFD, mux and charger drivers together with the mux
core required by the existing UART HDQ and BQ27xxx support.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: 606cf1cadc2881b3c5fd150d042defeb681dab6d
---
 arch/arm64/configs/defconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index a9a7a17d1..8fb097f84 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -485,6 +485,7 @@ CONFIG_CAN_RCAR_CANFD=m
 CONFIG_CAN_MCP251XFD=m
 CONFIG_MDIO_GPIO=m
 CONFIG_MDIO_BUS_MUX_MULTIPLEXER=y
+CONFIG_MUX_CORE=y
 CONFIG_MDIO_BUS_MUX_MMIOREG=y
 CONFIG_USB_PEGASUS=m
 CONFIG_USB_RTL8150=m
@@ -756,6 +757,7 @@ CONFIG_BATTERY_QCOM_BATTMGR=m
 CONFIG_BATTERY_SBS=m
 CONFIG_BATTERY_BQ27XXX=y
 CONFIG_BATTERY_BQ27XXX_HDQ=y
+CONFIG_CHARGER_APPLE_SN2400=y
 CONFIG_BATTERY_MAX17042=m
 CONFIG_CHARGER_MT6360=m
 CONFIG_CHARGER_BQ25890=m
@@ -852,6 +854,7 @@ CONFIG_MFD_RK8XX_I2C=y
 CONFIG_MFD_RK8XX_SPI=y
 CONFIG_MFD_SEC_ACPM=m
 CONFIG_MFD_SEC_I2C=y
+CONFIG_MFD_SN2400=y
 CONFIG_MFD_SL28CPLD=y
 CONFIG_RZ_MTU3=y
 CONFIG_MFD_TI_AM335X_TSCADC=m
@@ -1862,6 +1865,7 @@ CONFIG_OPTEE=y
 CONFIG_QCOMTEE=m
 CONFIG_MUX_GPIO=m
 CONFIG_MUX_MMIO=y
+CONFIG_MUX_SN2400=y
 CONFIG_SLIMBUS=m
 CONFIG_SLIM_QCOM_CTRL=m
 CONFIG_SLIM_QCOM_NGD_CTRL=m
